docs/docs/20-usage/20-pipeline-syntax.md

```diff
@@ -462,6 +462,13 @@ when:
   - evaluate: 'not (CI_COMMIT_MESSAGE contains "please ignore me")'
 ```
 
+Run on pull requests with the label `deploy`:
+
+```yaml
+when:
+  - evaluate: 'CI_COMMIT_PULL_REQUEST_LABELS contains "deploy"'
```

Comment:
As `CI_COMMIT_PULL_REQUEST_LABELS` is currently a string this only checks if the string contains `deploy`.

An exact match on a label would be better. Something clunky like:

`any([CI_COMMIT_PULL_REQUEST_LABELS], {value == "deploy"})`

Any ideas?

Reply:
expr has proper support for more complex data structures as our current string => string map. We could for example allow the user to directly access the metadata object. Problem I noticed when trying it, is that we would need to document it somehow.

If we stick to the env vars I would prefer something simple like the current approach and not introduce json strings which then need to be parsed again. Using `,` inside a label seems to be pretty uncommon to me.

Reply:
Would you both be fine with keeping this super simple for now by sticking to the string divided by `,` approach?

Reply:
@anbraten What about surrounding the string with `,` as well? i.e. `,ready for review,deploy,`

This would allow `contains ",deploy,"` to work every time.

Reply:
I'm fine with as is ... and the eval will have more info already and so should handle strange cases if you need to